torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1631,6 +1631,8 @@ def _init(
         self.fx_node_cache: Dict[Tuple[Callable, Tuple[Any, ...]], torch.fx.Node] = {}
         self.source_to_symbol: Dict[str, sympy.Symbol] = {}
 
+        self.source_to_symint_node_cache : Dict[TensorPropertySource, SymInt] = {}
```

Comment:
the type annotation and the variable name don't agree

Reply:
I think it's not good to store a SymInt here. The problem is that SymInt holds a strong reference to ShapeEnv, so this will cause a cycle that will prevent the ShapeEnv from ever getting deallocated. Thinking about how to deal with constants...

Reply:
ok, let's chat offline :)